### PR TITLE
feat: enable no-invalid-this

### PR DESCRIPTION
**Pull request by Prince-Mendiratta**
Context: https://github.com/Expensify/App/issues/17971

### CODE REVIEW

**Comment by s77rt** [comment]
No reviewers, No Assignees. Feels like a desert 🏜️ . Where is everybody

File: rules/es6.js [hunk 1]
```diff
@@ -23,5 +23,7 @@ module.exports = {
         // Do not use these features yet
         'es/no-nullish-coalescing-operators': 'error',
         'es/no-optional-chaining': 'error',
+
+        'no-invalid-this': 'error'
```
Comment (s77rt) [inline rules/es6.js:27]:
```suggestion
        'no-invalid-this': 'error',
```
This will fail on lint :sweat_smile: 

Can you also add a comment here?

**Review by s77rt — CHANGES_REQUESTED** [review]

**Review by s77rt — APPROVED** [review]
LGTM! :rocket: 

cc @tgolen

**Review by tgolen — APPROVED** [review]

**Comment by tgolen** [comment]
All right, this is published to npm now: eslint-config-expensify@2.0.36

**Pull request merged**